src/main/java/org/veupathdb/service/eda/ds/plugin/standalonemap/CollectionFloatingHistogramPlugin.java

```diff
@@ -0,0 +1,133 @@
+    if (pluginSpec.getBarMode() == null) {
+      throw new ValidationException("Property 'barMode' is required.");
+    }
+  }
+
+  @Override
+  protected List<StreamSpec> getRequestedStreams(CollectionFloatingHistogramSpec pluginSpec) {
+    return ListBuilder.asList(
+      new StreamSpec(DEFAULT_SINGLE_STREAM_NAME, pluginSpec.getOutputEntityId())
+        .addVars(pluginSpec.getOverlayConfig().getSelectedMembers())
+      );
+  }
+  
+  @Override
+  protected void writeResults(OutputStream out, Map<String, InputStream> dataStreams) throws IOException {
+    PluginUtil util = getUtil();
+    CollectionFloatingHistogramSpec spec = getPluginSpec();
+    List<VariableSpec> inputVarSpecs = new ArrayList<>(spec.getOverlayConfig().getSelectedMembers());
+    CollectionSpec overlayVariable = spec.getOverlayConfig().getCollection();
+    Map<String, CollectionSpec> varMap = new HashMap<String, CollectionSpec>();
+    varMap.put("overlay", overlayVariable);
+    String barMode = spec.getBarMode().getValue();
+    String collectionType = util.getCollectionType(overlayVariable);
+
+    useRConnectionWithRemoteFiles(Resources.RSERVE_URL, dataStreams, connection -> {
+      connection.voidEval(util.getVoidEvalFreadCommand(DEFAULT_SINGLE_STREAM_NAME, inputVarSpecs));
+      connection.voidEval(getVoidEvalCollectionMetadataList(varMap));
+     
+      String viewportRString = getViewportAsRString(spec.getViewport(), collectionType);
+      connection.voidEval(viewportRString);
+      
+      BinWidthSpec binSpec = spec.getBinSpec();
+      validateBinSpec(binSpec, collectionType);
+      String binReportValue = binSpec.getType().getValue() != null ? binSpec.getType().getValue() : "binWidth";
+      
+      String binWidth = "NULL";
+      if (collectionType.equals("NUMBER") || collectionType.equals("INTEGER")) {
+        binWidth = binSpec.getValue() == null ? "NULL" : "as.numeric('" + binSpec.getValue() + "')";
+      } else {
+        binWidth = binSpec.getValue() == null ? "NULL" : "'" + binSpec.getValue().toString() + " " + binSpec.getUnits().toString().toLowerCase() + "'";
+      }
+      connection.voidEval("binWidth <- " + binWidth);
+
+      String cmd =
+          "plot.data::histogram(data=" + DEFAULT_SINGLE_STREAM_NAME + ", " +
+                                  "variables=variables, " +
+                                  "binWidth=binWidth, " +
+                                  "value='" + spec.getValueSpec().getValue() + "', " +
+                                  "binReportValue='" + binReportValue + "', " +
+                                  "barmode='" + barMode + "', " +
+                                  "viewport=viewport, " + 
+                                  "sampleSizes=FALSE, " +
+                                  "completeCases=FALSE, " +
+                                  "overlayValues=NULL, " +
+                                  "evilMode='noVariables')";
+               System.err.println(cmd);
```

Comment:
This might be more appropriate as a `LOG.debug` or `LOG.trace`. I'm pretty sure stderr also goes to the docker logs, but it will look inconsistent with other log lines.

Reply:
sry sometimes i do that when im debugging a thing. ill just remove it.